packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1091,27 +1130,40 @@ function resolveExportsOrImports(
   })
 
   const fn = type === 'imports' ? imports : exports
-  const result = fn(pkg, key, {
+  const result = fn(pkg.data, key, {
     browser: targetWeb && !additionalConditions.has('node'),
     require: options.isRequire && !additionalConditions.has('import'),
     conditions,
   })
 
-  return result ? result[0] : undefined
+  if (!result) {
+    return undefined
+  }
+
+  // We need to support array conditions like `["./*.js", "./*.mjs"]`
+  // So loop through each result and find one that actually exists
+  if (result.length > 1) {
+    for (const entry of result) {
+      if (fs.existsSync(path.join(pkg.dir, entry))) {
+        return entry
+      }
+    }
+  }
```

Comment:
We currently intentionally don't support this as it's not following the spec. Node.js only specifies to use the next element if the specifier is invalid, not if it exists on the fs.

Reply:
Good to know, I can remove it.